utils/file.go

```diff
@@ -21,11 +21,12 @@ import (
 	"io/fs"
 	"os"
 	"path/filepath"
+	"runtime"
 )
 
 // SaferWriteFile writes to a temporary file and then replaces the expected output file.
 // This prevents other processes from reading partial content while the writer is still writing.
-func SaferWriteFile(content []byte, outputFile string, perm fs.FileMode) error {
+func SaferWriteFile(content []byte, outputFile string, perm fs.FileMode, uid, gid int) error {
```

Comment:
Please change it to `SaferWriteFile(content []byte, outputFile string, opts FileOptions)` where `FileOptions` would be:

```golang
type FileOptions struct {
  Perm fs.FileMode
  UID *int
  GID *int
}
```

That way the current caller would look like:

```golang
opts := FileOption{Perm: 0644}
if err := SaferWriteFile(data, filePath, opts); err != nil {
...
```

And the ones that need to change GID, UID would look like:

```golang
opts := FileOption{Perm: 0644, GID: myGID, UID: myUID}
if err := SaferWriteFile(data, filePath, opts); err != nil {
...
```

And where you are calling `Chown()` you can do:

```golang
preserveGUIDs = -1
GID := preserveGUIDs
if opts.GID != nil {
  GID = *opts.GID
}

UID := preserveGUIDs
if opts.ID != nil {
  UID = *opts.UID
}

if err := os.Chown(tmp.Name(), UID, GID); err != nil && runtime.GOOS != "windows" {
...
```

Reply:
Done. Also added unit testing for the UID and GID options.